src/osd/OSD.h

```diff
@@ -2527,4 +2527,12 @@ extern const CompatSet::Feature ceph_osd_feature_compat[];
 extern const CompatSet::Feature ceph_osd_feature_ro_compat[];
 extern const CompatSet::Feature ceph_osd_feature_incompat[];
 
+namespace ceph { 
+namespace osd_cmds { 
+
+int heap(CephContext& cct, cmdmap_t& cmdmap, Formatter& f, std::ostream& os);
```

Comment:
I agree, will do. I think ultimately the "real" right solution is to split these into their own source file away from the other OSD.cc code, but that's a much more significant change.